
message FfmpegControl
{
    enum Opcode
    {
        NONE = 0;
        RECORD = 1;
        IDLE = 2;
        PAUSE = 3;
        UNPAUSE = 4;
        SHUTDOWN = 5;
    }

    optional Opcode opcode = 1;

    // capture screen area
    optional int32 capture_x = 6;
    optional int32 capture_y = 7;
    optional int32 capture_width = 8;
    optional int32 capture_height = 9;

    // capture fps
    optional int32 capture_fps = 10;

    // capture devices
    optional string audio_device = 11;
    optional string video_device = 12;

    // display recorded video in a window as it is being recorded
    optional bool debug_show_video = 13;

    // recorded video output filename
    optional string output_file = 14;

    // scale the recorded video size, e.g. scale=0.5 will store the recorded
    // video half the captured size
    optional double scale = 15;
}

message FfmpegStatus
{
    optional bool is_recording = 1;
    optional bool is_paused = 2;
    optional bool has_crashed = 3;
}

